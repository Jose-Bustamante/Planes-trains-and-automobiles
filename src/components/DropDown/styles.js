import styled from "styled-components";
import FormControl from "@material-ui/core/FormControl";

export const StyledFormControl = styled(FormControl)`
  &&& {
    margin: 8px;
    min-width: 120px;
    border: 0;
    display: inline-flex;
    padding: 0;
    position: relative;
    flex-direction: column;
    vertical-align: top;
  }
`;
