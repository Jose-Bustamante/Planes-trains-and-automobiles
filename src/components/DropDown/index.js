import React from "react";
import PropTypes, { string } from "prop-types";
import Select from "@material-ui/core/Select";
import MenuItem from "@material-ui/core/MenuItem";
import InputLabel from "@material-ui/core/InputLabel";
import { StyledFormControl } from "./styles";
const DropDown = ({ options, name, onSelect, onClean, value }) => {
  const handleChange = (event) => {
    if (event.target.value === "") {
      onClean(name);
    } else {
      onSelect({
        filterBy: name,
        value: event.target.value,
      });
    }
    event.preventDefault();
  };

  if (options.length) {
    return (
      <StyledFormControl>
        <InputLabel id={name}>{name}</InputLabel>

        <Select
          labelId={name}
          id="select"
          value={value}
          onChange={handleChange}
          label={name}
          inputProps={{
            id: `input-${name}`,
            "data-testid": `input-${name}`,
          }}
        >
          <MenuItem value="">
            <em>None</em>
          </MenuItem>
          {options.map((option) => (
            <MenuItem key={option} value={option}>
              {option}
            </MenuItem>
          ))}
        </Select>
      </StyledFormControl>
    );
  }
  return null;
};

DropDown.propTypes = {
  options: PropTypes.arrayOf(string), // array of available options
  name: PropTypes.string, // name used to distinguish the dropDown
  onSelect: PropTypes.func, // callback when option is selected
  onClean: PropTypes.func, // callback when default option is selected
};

DropDown.defaultProps = {
  name: "",
  options: [],
};

DropDown.displayName = "DropDown";

export default DropDown;
