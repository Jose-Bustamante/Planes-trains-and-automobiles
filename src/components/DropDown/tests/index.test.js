import React from "react";
import { render } from "@testing-library/react";

import DropDown from "../index";

describe("<DropDown />", () => {
  const options = ["car", "train", "plane"];
  const onSelect = jest.fn();
  it("renders nothing since no options are passed", () => {
    const { asFragment } = render(<DropDown name="type" onSelect={onSelect} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders DropDown with defined values", () => {
    const { asFragment } = render(
      <DropDown options={options} name="type" onSelect={onSelect} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
