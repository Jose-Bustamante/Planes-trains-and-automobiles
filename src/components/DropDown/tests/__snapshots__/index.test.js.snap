// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDown /> renders DropDown with defined values 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root sc-bdnylx iSWsxH"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
      id="type"
    >
      type
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="type select"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="select"
        role="button"
        tabindex="0"
      >
        <span>
          â€‹
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        data-testid="input-type"
        id="input-type"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<DropDown /> renders nothing since no options are passed 1`] = `<DocumentFragment />`;
